src/terminal/input/terminalInput.cpp

```diff
@@ -552,16 +508,16 @@ bool TerminalInput::HandleKey(const IInputEvent* const pInEvent)
         {
             _lastVirtualKeyCode = std::nullopt;
         }
-        return false;
+        return MakeUnhandled();
     }
 
     // If this is a repeat of the last recorded key press, and Auto Repeat Mode
     // is disabled, then we should suppress this event.
     if (matchingLastKeyPress && !_inputMode.test(Mode::AutoRepeat))
     {
-        // Note that we must return true here to say we've handled the event,
-        // otherwise the key press can still end up being submitted.
-        return true;
+        // Note that we must return an empty string here to imply that we've handled
+        // the event, otherwise the key press can still end up being submitted.
+        return MakeOutput({});

```

Comment:
To be clear `MakeOutput({})` returns the empty string, whereas `MakeUnhandled()` returns `OutputType{}`, which is... what exactly? A null string?

Reply:
Yeah kind of. It's like returning `null` vs. `""` in JS. `null` in this case means "we didn't handle it", where `""` means we did handle it and the result is simply an empty string.